Apps/WebClient/src/NewClientApp/src/components/modal/AddQuickLinkComponent.vue

```diff
@@ -0,0 +1,319 @@
+
+    return userStore.setUserPreference(preference).then(() => {
+        selectedQuickLinks.value = selectedQuickLinks.value.filter(
+            (link) => link !== quickLinkName
+        );
+    });
+}
+
+async function handleSubmit(): Promise<void> {
+    clearErrors();
+
+    selectedQuickLinks.value.forEach((element) =>
+        logger.debug(`Adding quick link:  ${element}`)
+    );
+
+    const links: QuickLink[] = quickLinks.value ?? [];
+    selectedQuickLinks.value.forEach((module) => {
+        const details = getEntryTypeByModule(module);
+        if (details) {
+            links.push({
+                name: details.name,
+                filter: { modules: [module] },
+            });
+        }
+    });
+
+    try {
+        isLoading.value = true;
+
+        const promises = [
+            userStore.updateQuickLinks(user.value.hdid, links),
+            updateSelectedUserPreference(
+                "organ-donor-registration",
+                UserPreferenceType.HideOrganDonorQuickLink
+            ),
+            updateSelectedUserPreference(
+                "bc-vaccine-card",
+                UserPreferenceType.HideVaccineCardQuickLink
+            ),
+            updateSelectedUserPreference(
+                "immunization-record",
+                UserPreferenceType.HideImmunizationRecordQuickLink
+            ),
+        ].filter((p) => p !== undefined);
+
+        await PromiseUtility.withMinimumDelay(Promise.all(promises), 100);
+
+        // Hide the modal manually
+        await nextTick();
+        hideModal();
+    } catch (error) {
+        if (isTooManyRequestsError(error)) {
+            errorStore.setTooManyRequestsError("addQuickLinkModal");
+        } else {
+            bannerError.value = ErrorTranslator.toBannerError(
+                ErrorType.Update,
+                ErrorSourceType.QuickLinks,
+                undefined
+            );
+        }
+    } finally {
+        // Clear selected quick links
+        selectedQuickLinks.value = [];
+
+        isLoading.value = false;
+    }
+}
+
+function hideModal(): void {
+    isVisible.value = false;
+}
+</script>
+
+<template>
+    <v-dialog
+        id="add-quick-link-modal"
+        v-model="isVisible"
+        data-testid="add-quick-link-modal"
+        persistent
+    >
+        <template v-slot:activator="{ props }">
+            <HgButtonComponent
+                id="add-quick-link-button"
+                data-testid="add-quick-link-button"
+                :disabled="disabled"
+                class="float-right"
+                variant="secondary"
+                v-bind="props"
+            >
+                <v-icon icon="plus" size="medium" class="mr-2" />
+                <span class="text-body-1">Add Quick Link</span>
+            </HgButtonComponent>
+        </template>
+        <v-row justify="center" data-testid="quick-link-modal-text">
+            <v-card min-width="20%">
+                <v-card-title class="bg-primary text-white">
```

Comment:
```suggestion
                <v-card-title class="bg-primary text-white px-0">
```
There's double padding being added by both the title and the toolbar - removing the horizontal padding from the title makes it look pretty decent:

![image](https://github.com/bcgov/healthgateway/assets/16570293/1e73b2f1-09b8-443d-805e-499ed344f927)